src/main/java/com.company/tree/BinarySearchTreeValidator.java

```diff
@@ -0,0 +1,39 @@
+package com.company.tree;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+
+public class BinarySearchTreeValidator {
+    public boolean validateBinarySearchTree(TreeNode root) {
+        return validateBST(root, new Range(Long.MIN_VALUE, Long.MAX_VALUE));
+    }
+
+    private boolean validateBST(TreeNode root, Range range) {
+        if (root == null) {
+            return true;
+        }
+        if (root.val <= range.min || root.val >= range.max) {
+            return false;
+        }
+        return validateBST(root.left, new Range(root.val, range.max)) && validateBST(root.right, new Range(range.min, root.val));
+    }
+
+    @NoArgsConstructor
+    @AllArgsConstructor
+    class Range {
+        long min;
```

Comment:
If you are doing direct to min and max variable, you can keep these value as public.

Reply:
changed to private